ocaml/xenopsd/xc/device.ml

```diff
@@ -1238,7 +1240,57 @@ module PCI = struct
                 shift_right_logical (add _page_size scan_size |> pred) 12
               )
             in
+            (* Linux disables the ROM BARs if not used and by default does
+               not set the start address.
+               Force it to set the address using "rom" file.
+               This method works also with lockdown mode enabled. *)
+            let enable_rom start =
+              (* read current configured ROM address to check if enabled *)
+              let config_fn = sysfs_pci_dev ^ "/config" in
+              let out = Bytes.make 4 '\x00' in
+              let fd =
+                Unix.openfile config_fn [Unix.O_RDONLY; Unix.O_CLOEXEC] 0
+              in
+              let current_addr =
+                finally
+                  (fun () ->
+                    Unix.(lseek fd 0x30 SEEK_SET) |> ignore ;
+                    (* this is a virtual file that control PCI configuration,
+                       it will either fail or return 4 *)
+                    Unix.(read fd out 0 4) |> ignore ;
+                    Nativeint.(
+                      logand (of_int32 (Bytes.get_int32_ne out 0)) 0xfffff800n
+                    )
+                  )
+                  (fun () -> Unix.close fd)
+              in
+              if current_addr <> start then (
+                (* to enable output "1" on "rom" file and try to read it *)
+                let rom_fn = sysfs_pci_dev ^ "/rom" in
+                let fd =
+                  Unix.openfile rom_fn [Unix.O_WRONLY; Unix.O_CLOEXEC] 0
```

Comment:
I notice that all errors are ignored and none are logged. So how can we understand what is happening?